test/integration/service/loadbalancer_test.go

```diff
@@ -353,6 +349,64 @@ func Test_ServiceLoadBalancerEnableLoadBalancerClassThenUpdateLoadBalancerClass(
 	}
 }
 
+// Test_ServiceUpdateLoadBalancerEnableLoadBalancerClass tests that when a Service that is updated
+// from ClusterIP to LoadBalancer with loadBalancerClass set, cloud provider should not
+// create load balancer.
+func Test_ServiceUpdateLoadBalancerEnableLoadBalancerClass(t *testing.T) {
+
+	controlPlaneConfig := framework.NewIntegrationTestControlPlaneConfig()
+	_, server, closeFn := framework.RunAnAPIServer(controlPlaneConfig)
+	defer closeFn()
+
+	config := restclient.Config{Host: server.URL}
+	client, err := clientset.NewForConfig(&config)
+	if err != nil {
+		t.Fatalf("Error creating clientset: %v", err)
+	}
+
+	ns := framework.CreateTestingNamespace("test-service-immutable-load-balancer-class", server, t)
+	defer framework.DeleteTestingNamespace(ns, server, t)
+
+	controller, cloud, informer := newServiceController(t, client)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	informer.Start(ctx.Done())
+	go controller.Run(ctx, 1)
+
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-load-balancer-class",
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{{
+				Port: int32(80),
+			}},
+		},
+	}
+
+	service, err = client.CoreV1().Services(ns.Name).Create(ctx, service, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Error creating test service: %v", err)
+	}
+
+	if len(cloud.Calls) > 0 {
```

Comment:
Aren't we expecting cloud.Calls > 0 here since we didn't set load balancer class?

Reply:
The service type is `clusterIP` here, this case tests update from other service types to `loadBalancer` and set `loadBalancerClass`. 😊

Reply:
I have updated the test cases to update from a `loadBalancer` service to `loadBalancer` with `loadBalancerClass` set.